griptape/drivers/vector/mongodb_vector_store_driver.py

```diff
@@ -135,3 +135,12 @@ def query(
         ]
 
         return results
+
+    def delete_vector(self, vector_id: str):
+        """Deletes the vector from the collection."""
+        collection = self.get_collection()
+        collection.delete_one({"_id": vector_id})
+
+    def delete_collection(self):
+        """Deletes the current collection from the database."""
+        self.get_collection().drop()
```

Comment:
Should the collection really be deleted here? If so, let's not move it into a separate method since it's a one-liner not used anywhere else.

Reply:
I don't fully understand.
First of all, yes, the objective of this function is to delete the collection.
But where should I move it?

Reply:
Sorry, I misread the code. I don't think we need this method at all. The purpose of the driver is to handle vectors inside of a vector DB, not the DB itself.

Reply:
I, for once, need that functionality with our current use case, where we create embeddings of large documents and store them in individual collections.
Also, check out the screenshot attached. When you upload a big document as an embedding, it makes a bunch of vectors for each embedding. So, this is where the 'delete_collection' function comes in handy
![image](https://github.com/griptape-ai/griptape/assets/50590409/6b1158ff-1bf7-4e19-a6b9-bfa33b4dacd4)

Reply:
Your use case makes perfect sense but we are trying to keep all drivers pretty generic.

In this particular scenario, the user could chose between any number of ways to store vectors (e.g., single collection, collection per artifact, collection per artifact grouping, etc.), which could result in different `delete_*` functionality. We are also, in general, trying to reduce the number of one-off methods in drivers that could result in inconsistent UX

In your use case, would it make more sense to just use the Mongo SDK to delete the collection instead of moving that functionality into Griptape?

Reply:
I understand the UX aspect, but Griptape also creates the collection, therefore, it should provide a functionality to delete them as well, without going the way of using pymongo or any other SDK. 

Also, maybe I misunderstood you there, but there is no "out of the box" solution from griptape to store the vectors in the fashion that you just described. 
Furthermore, as I said above, when creating an embedding from a large document, you cannot store the entire thing in one vector, there will be several vectors created for that embedding, therefore, the delete_vector function would probably result in some weird work arounds to properly delete the entire embedding.

But this is my opinion - if you guys agree that the delete_collection function is not of use for griptape, I will remove it from the PR.